src/System.Net.HttpListener/tests/GetContextHelper.cs

```diff
@@ -60,7 +60,15 @@ public class Helpers
     {
         public static void WaitForSocketShutdown(Socket socket)
         {
-            while (SocketConnected(socket));
+            if (PlatformDetection.IsWindows || PlatformDetection.IsOSX)
+            {
+                socket.Shutdown(SocketShutdown.Both);
```

Comment:
On my Ubuntu VM these tests all hung without this code. I added OSX because I suspected that's what caused the OSX failures. (I was wrong)

I'm also now a bit nervous about these tests, as i dont know if they're going to fail in the CI due to timing etc
I've ran the tests on windows 1000 times and nothing failed but I'm not sure about ubuntu so I will test that before merging. Seems like the CI hates me